### ISSUE TITLE
Trace is not outputted when `--output=table`

### ISSUE THREAD

**Issue by tonglil** [open-policy-agent/conftest#1031]
~As of v0.55.0, the CONFTEST_TRACE env var does not seem to set the `conftest test --trace` flag.~

I expected the trace to still be displayed even if the output is set to something else.

**Comment by thevilledev** [comment]
Could not reproduce this on v0.55.0 or v0.56.0 (latest) on macOS 15.2. These commands produce identical output, sometimes in different order:

```
$ CONFTEST_TRACE=true conftest test examples/kubernetes/service.yaml -p examples/kubernetes/policy

$ conftest test --trace examples/kubernetes/service.yaml -p examples/kubernetes/policy
```

Looking at the code:
-  `internal/commands/default.go` sets the env var prefix correctly
- `internal/commands/test.go` binds flagname `trace` correctly
- `TestRunner` struct in `runner/test.go` has a Trace field, and Viper binds struct fields to variables case-insensitively. Even though the struct tag `mapstructure:"trace"` is missing I think this should work.

Which environment was this failing on?

**Comment by tonglil** [comment]
Thanks for diving into this @thevilledev.

Upon closer inspection, I was setting the output flag to `table`, which seems to override `trace`, which I did not expect and falsely attributed to the flag not working.

**Comment by jalseth** [comment]
I agree this is not intuitive (or documented) behavior. I haven't had time to dig into why it is this way, so there may be a reason. PRs are welcome.

**Comment by thevilledev** [comment]
I noticed the tracing flag is only available for standard output. It's not passed to other output formats even if defined, on the other hand why would it be. I have a proposal to fix this in #1060 so that if the flag is defined, output format always defaults to standard.

**Comment by thevilledev** [comment]
PR #1060 reduced to a documentation update due to concerns of making this a breaking change.

Here's my proposal to support tracing for any output format:

## Problem
Currently, users cannot use tracing together with custom output formats (JSON, table, etc.) because the output format takes priority. This limits debugging capabilities when working with tools that require specific output formats.

## Proposed Solution
Allow `--trace` to work with any output format by:
- Writing trace output to stderr
- Writing the formatted output (JSON, table, etc.) to stdout

This way:
- Tools can still parse their expected output format from stdout
- Users can see trace information for debugging in stderr
- No breaking changes to existing behavior

Example desired usage:
```console
# Trace output goes to stderr, JSON output to stdout
$ conftest test --trace --output json deployment.yaml 2>trace.log
```

## Implementation Notes
- The `Standard` outputter would need to be modified to support writing trace output to a different stream than the main output
- Need to ensure cross-platform compatibility (Windows, Unix-like systems)
- Consider adding a flag to control trace output destination (stdout/stderr/file)

## Benefits
- Better debugging experience when using tools that require specific output formats
- No need to switch between trace mode and output format mode
- Follows standard practice of using stderr for diagnostic output
